### ISSUE TITLE
allow optional backward-compatible repeating groups in sbe 1.0

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [real-logic/simple-binary-encoding#193]
The following conversation:

http://www.fixtradingcommunity.org/pg/discussions/topicpost/1228257/

seems to imply that an sbe 1.0 header format change is needed to allow optional repeating groups
while preserving message backward-compatibility, and it has to happen only later, in sbe 2.0.

However it seems that all is needed is to allow optional repeating groups in sbe 1.0
is a support for `sinceVersion="N"` and `presence="optional"` for the `groupType`,
similar to how it is done for optional fields.

https://github.com/real-logic/simple-binary-encoding/blob/master/main/resources/fpl/SimpleBinary1-0.xsd#L108

In other words, "next" version decoder will get a "null/empty" group object 
when decoding the "past" version message (which does not have a new optional group), 
w/o affecting the offsets.
